src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -656,16 +685,35 @@ public void saveResolutions()
     Compression compression = N5Compression.getCompressor(compressionType,
             compressionParameter);
 
-    N5Writer n5 = new N5FSWriter(outputPath.resolve(pyramidName).toString());
+    final N5Writer n5;
```

Comment:
Maybe resolve the path once outside conditionals and construct with switch if we use a file type string above?